src/vs/base/common/lifecycle.ts

```diff
@@ -430,6 +430,34 @@ export class DisposableStore implements IDisposable {
 
 		return o;
 	}
+
+	/**
+	 * Deletes a disposable from store and disposes of it. This will not throw or warn and proceed to dispose the
+	 * disposable even when the disposable is not part in the store.
+	 */
+	public delete<T extends IDisposable>(o: T): void {
```

Comment:
I know it is called `Set.delete`, but I think this method should be called `remove`. Better even would be `removeAndDispose`

Reply:
I was going to call it remove, but it's using add and delete internally on Set which aligns with the web api. Instead of delete and deleteAndDispose, I flipped them do delete (and dispose, common case) and deleteAndLeak (rare case).

Reply:
I just think it might be surprising that `delete` also disposes the object.

Reply:
I would think the opposite, the main concern is leaks so disposing is the safe option imo.

Reply:
We can be explicit on both though, as long as the leak one is called out